services/director/source/registry_proxy.py

```diff
@@ -86,3 +86,44 @@ def get_service_sub_name(repository_name):
     if last_suffix_index < 0:
         raise Exception('Invalid service name: ' + repository_name)
     return list_of_suffixes[last_suffix_index]
+
+def _get_repo_details(repo):
+    #pylint: disable=too-many-nested-blocks
+    current_repo = {}
+    if "/comp/" in repo:
```

Comment:
is it necessary to use inband signalling here ? can't we represent the repo as a map with relevant properties ?

Reply:
you mean the fact that I filter for comp? This is only temporarily because at the moment there are services in the platform that need multiple images from the registry. We have to clean this up

Reply:
ok